guide/twilio_sample.bal

```diff
@@ -0,0 +1,192 @@
+        P{{config}} Twilio configuraion
+    }
+    public function init(TwilioConfiguration config);
+
+    documentation { Initialize Twilio endpoint
+        R{{}} The Twilio connector object
+    }
+    public function getCallerActions() returns TwilioConnector;
+
+};
+
+// Configuration
+documentation {
+    F{{clientConfig}} The http client endpoint for basic configuration
+}
+public type TwilioConfiguration {
+    http:ClientEndpointConfig clientConfig;
+};
+
+// Connector
+documentation {Object to initialize the connection with Twilio.
+    F{{accountSId}} Unique identifier of the account
+    F{{client}} Http client endpoint for basic api
+}
+public type TwilioConnector object {
+
+    public {
+        string accountSId;
+        http:Client client;
+    }
+
+    documentation { Return account details of the given account-sid
+        R{{}} If success, returns account object with basic details, else returns TwilioError object
+    }
+    public function getAccountDetails() returns (Account|error);
+};
+
+//Record to represent type
+documentation {
+    F{{sid}} Unique identifier of the account
+    F{{name}} The name of the account
+    F{{status}} The status of the account (active, suspended, closed)
+    F{{^"type"}} The type of this account (Trial, Full)
+    F{{createdDate}} The date that this account was created
+    F{{updatedDate}} The date that this account was last updated
+}
+public type Account {
+    string sid;
+    string name;
+    string status;
+    string ^"type";
+    string createdDate;
+    string updatedDate;
+};
+
+// Constants
+@final string BASE_URL = "https://api.twilio.com/2010-04-01";
+@final string ACCOUNTS_API = "/Accounts/";
+@final string RESPONSE_TYPE_JSON = ".json";
+@final string EMPTY_STRING = "";
+
+// =========== Implementation of the Endpoint
+public function TwilioClient::init(TwilioConfiguration config) {
+
+    config.clientConfig.url = BASE_URL;
+    string username;
+    string password;
+
+    var usernameOrEmpty = config.clientConfig.auth.username;
+    match usernameOrEmpty {
+        string usernameString => username = usernameString;
+        () => {
+            error err;
+            err.message = "Username cannot be empty";
+            throw err;
+        }
+    }
+
+    var passwordOrEmpty = config.clientConfig.auth.password;
+    match passwordOrEmpty {
+        string passwordString => password = passwordString;
+        () => {
+            error err;
+            err.message = "Password cannot be empty";
+            throw err;
+        }
+    }
+
+    self.twilioConnector.accountSId = username;
+    self.twilioConnector.client.init(config.clientConfig);
+}
+
+public function TwilioClient::getCallerActions() returns TwilioConnector {
+    return self.twilioConnector;
+}
+// =========== End of implementation of the Endpoint
+
+
+// =========== Implementation for Connector
+public function TwilioConnector::getAccountDetails() returns (Account|error) {
```

Comment:
Shall we remove the parentheses around the return type when it is a union type (here and in other similar functions)? 
```bal
public function TwilioConnector::getAccountDetails() returns Account|error
```

As a practice we use parentheses with return types only when required with tuple types.

Reply:
Thanks. Fixed.